shared/src/akri/configuration.rs

```diff
@@ -18,6 +20,30 @@ use std::collections::HashMap;
 
 pub type ConfigurationList = ObjectList<Configuration>;
 
+/// This defines kinds of discovery property source
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub enum DiscoveryPropertySource {
+    /// Source is a key of a ConfigMap.
+    ConfigMapKeyRef(ConfigMapKeySelector),
+    /// Source is a key of a Secret.
+    SecretKeyRef(SecretKeySelector),
+}
+
+/// DiscoveryProperty represents a property for discovery devices
+#[derive(Serialize, Deserialize, Clone, Debug, JsonSchema, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct DiscoveryProperty {
+    /// Name of the discovery property
+    pub name: String,
+
+    /// value of the discovery property
+    pub value: Option<String>,
```

Comment:
`#[serde(default, skip_serializing_if = "Option::is_none")]` ?

Reply:
for value and value_from?

Reply:
add serde attributes for value and value_from